student-work/jeff_beyer/exam/exam.py

```diff
@@ -0,0 +1,73 @@
+import doctest
+import json
+import pandas as pd
+
+'''
+Exam submission for Pythonii - Jeff Beyer - 08/30/2017
+'''
+
+
+def read_exam(csv_location='exam.csv'):
+    '''
+    read_exam will pull the exam.csv file from either an file path or a URL.
+    If no input is given, it will default to looking for exam.csv in the
+    current folder.
+
+    Doctest if an invalid path is given:
+
+    >>> read_exam('invalid_file.csv')
+    ('', '', '')
+    '''
+    try:
+        # Read the csv as a data frame
+        df = pd.read_csv(csv_location)
+        # Pull the desired columns
+        info = (list(df['question number']), list(df['question']),
+                list(df['answer']))
+    except:
+        # If we didn't read anything, return an empty 3-tuple
+        info = ('', '', '')
+    return info
+
+
+def setup():
+    '''
+    Doctests for checking the length of the 4 lists:
+
+    >>> len(QUESTIONS)
+    11
+    >>> len(ANSWERS)
+    11
+    >>> len(QUESTION_TEXTS)
+    11
+    >>> len(MY_ANSWERS)
+    11
+    '''
+    # Read the exam file from web URL
+    exam_info = read_exam('https://raw.githubusercontent.com/hackoregon/' +
+                          'civicu-pythonii-summer-2017/ceasar/lessons/' +
+                          'shared-resources/exam.csv')
+    # Read the exam file from the local directory
+    # exam_info = read_exam('exam.csv')
+    # Read the exam file (without input)
+    # exam_info = read_exam()
+
+    # Store the exam info as global variables
+    globals()['QUESTIONS'] = exam_info[0]
```

Comment:
Nice! Love the use of `globals()` and reuse of your `read_exam` function. FYI a neat python pattern is to assign a tuple to 2 or more variables all at once (you just must have the same number of assigned variables as the length of the tuple), like this:

`globals()['QUESTIONS'], globals()['QUESTION_TEXTS'], globals()['ANSWERS'] = exam_info`